floss/main.py

```diff
@@ -349,19 +357,24 @@ def select_functions(vw, asked_functions: Optional[List[int]]) -> Set[int]:
     return asked_functions_
 
 
-def is_supported_file_type(sample_file_path: Path):
+def get_file_type(sample_file_path: Path):
     """
-    Return if FLOSS supports the input file type, based on header bytes
+    Returns input file type, based on header bytes
     :param sample_file_path:
-    :return: True if file type is supported, False otherwise
+    :return: file type if it is supported, raise Error Otherwise
     """
     with sample_file_path.open("rb") as f:
-        magic = f.read(2)
+        magic = f.read(4)
 
-    if magic in SUPPORTED_FILE_MAGIC:
-        return True
+    if magic == SUPPORTED_FILE_MAGIC_ELF:
+        return SUPPORTED_FILE_MAGIC_ELF
+    elif magic[:2] == SUPPORTED_FILE_MAGIC_PE:
+        return SUPPORTED_FILE_MAGIC_PE
     else:
-        return False
+        raise WorkspaceLoadError(
+            "FLOSS currently supports the following formats for string decoding and stackstrings: PE,ELF\n"
+            "You can analyze shellcode using the --format sc32|sc64 switch. See the help (-h) for more information."
+        )
```

Comment:
need to raise this elsewhere so we can still handle shell code (without supported magic)

otherwise, this looks good to go, overall

Reply:
Just returned None for file type other than ELF,PE and rest logic is same.